### PR TITLE
Implemented alltrue, any, array_equal

### PR DESCRIPTION
**Pull request by MohamedElgharbawy**
Also removed implemented functions from `api_coverage.py`.

### CODE REVIEW

**Comment by MohamedElgharbawy** [comment]
Looking into the Travis errors.

File: tests/numpy/test_misc.py [hunk 1]
```diff
@@ -121,12 +121,102 @@ def test_all(nps_app_inst):
     assert nps.all(false_float).dtype is np.bool
 
 
+def test_alltrue(nps_app_inst):
```
Comment (elibol) [inline tests/numpy/test_misc.py:124]: Is `alltrue` the same as `all`? If so, I think it's okay to test `all` and `alltrue` in the same test.

File: tests/numpy/test_misc.py [hunk 2]
```diff
@@ -121,12 +121,102 @@ def test_all(nps_app_inst):
...
+
+    assert nps.alltrue(true_int).dtype is np.bool
+    assert nps.alltrue(false_int).dtype is np.bool
+    assert nps.alltrue(true_bool).dtype is np.bool
+    assert nps.alltrue(false_bool).dtype is np.bool
+    assert nps.alltrue(true_float).dtype is np.bool
+    assert nps.alltrue(false_float).dtype is np.bool
+
+
+def test_any(nps_app_inst):
```
Comment (elibol) [inline tests/numpy/test_misc.py:152]: This looks very similar to `all` and `alltrue`. Can we consolidate the code for these tests?

**Review by elibol — COMMENTED** [review]
Looks good, and thanks for cleaning up `api_coverage.py`! Please see comments.

File: tests/numpy/test_misc.py (hunk 1 above)
Comment (MohamedElgharbawy) [reply tests/numpy/test_misc.py:124]: Yes, I think `all` replaced `alltrue` in NumPy, but both are still supported. I'll combine their tests.

File: tests/numpy/test_misc.py (hunk 2 above)
Comment (MohamedElgharbawy) [reply tests/numpy/test_misc.py:152]: Yep!

File: nums/numpy/api.py [hunk 3]
```diff
@@ -398,6 +410,11 @@ def allclose(a: BlockArray, b: BlockArray, rtol=1.e-5, atol=1.e-8, equal_nan=Fal
     return _instance().allclose(a, b, rtol, atol)
 
 
+def array_equal(a1: BlockArray, a2: BlockArray, equal_nan=False) -> BlockArray:
```
Comment (elibol) [inline nums/numpy/api.py:413]: I think this implementation is okay, but please test the semantics of `array_equal` rigorously to ensure that `allclose` with 0 tolerance is an equivalent function to `array_equal`. Alternatively, we can discuss how to generalize the `allclose` function in application to support both `allclose` and `array_equal`.

Comment (MohamedElgharbawy) [reply nums/numpy/api.py:413]: `allclose` uses the following equation: `absolute(a - b) <= (atol + rtol * absolute(b))`. It's typically used for float arrays to prevent floating point error, while `array_equal` is meant for integer arrays. Keeping those usage cases in mind, I believe this implementation should be okay, right? I have a test in `test_array_equal` that has a very small difference in floating points, and `array_equal` returns false as expected.

Comment (elibol) [reply nums/numpy/api.py:413]: Okay. To be safe, please test edge cases such as positive/negative zero, behavior for pos/neg infinity, and comparisons between every combination of bool, int, float, and complex types (e.g. is `False` equal to `0.0`).

Comment (MohamedElgharbawy) [reply nums/numpy/api.py:413]: Will do.
